### PR TITLE
Fix #275218: MSVC warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Disable MSVC warnings for thirdparty code:

* C4267 for beatroot
* C4244 and C4267 for freetype
* C4267 and C4334 for poppler (in addition to those disabled earlier)
* C4028, C4189, C4267, C4311 and C4312 for portmidi
* C4267 for rtf2html

Fix more MSVC warnings:

* The one C4100
* The two C4457
* The two C4701
* The 475 C4267 

Also fix GCC warnings reg. wrong printf format for a size_t, reg. unused variables and reg. ambigous else branch, seen on Travis CI

Leaves us with (building for 64bit RelWithDebInfo):

* 193 C4456
* 122 C4458
* 1 LNK4098

### CODE REVIEW

File: libmscore/shape.h [hunk 1]
```diff
@@ -71,7 +71,7 @@ class Shape : public std::vector<ShapeElement> {
       qreal top() const;
       qreal bottom() const;
 
-      int size() const   { return std::vector<ShapeElement>::size(); }
+      int size() const   { return int(std::vector<ShapeElement>::size()); }
```
Comment (anatoly-os) [inline libmscore/shape.h]: I would use `size_t` instead as a return value.

File: libmscore/figuredbass.h [hunk 2]
```diff
@@ -288,7 +288,7 @@ class FiguredBass final : public TextBase {
                                                           return 0;   }
       qreal             printedLineLength() const     { return _printedLineLength; }
       bool              onNote() const          { return _onNote; }
-      int               numOfItems() const      { return items.size(); }
+      int               numOfItems() const      { return int(items.size()); }
```
Comment (anatoly-os) [inline libmscore/figuredbass.h]: See below.

File: libmscore/shape.h (hunk 1 above)
Comment (Jojo-Schmitz) [reply libmscore/shape.h]: That was my first idea and attempt too, but that leads to tons of further warnings sat places where the result of these methods are used and compared with other values

Comment (anatoly-os) [reply libmscore/shape.h]: ouch... Let's add the //TODO comment here with the proposal to replace all the casts to `size_t`. Once, we have to do it.

Comment (Jojo-Schmitz) [reply libmscore/shape.h]: Maybe. OTOH these casts just do what the compiler implicitly does anyhow.

File: libmscore/figuredbass.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply libmscore/figuredbass.h]: or above?

File: libmscore/shape.h (hunk 1 above)
Comment (Jojo-Schmitz) [reply libmscore/shape.h]: Seems I was wrong, probably die to missing the tree in the forest, `size_t` is possible here

**Comment by Jojo-Schmitz** [comment]
Neecd to investigate why mtest tst_copypaste crashes and tst_splitstaff fails now

**Comment by Jojo-Schmitz** [comment]
Need to investigate why mtest tst_copypaste crashes and tst_splitstaff fails now.
Interesting that the 1st commit now fails too, after a rebase, very strange, as that does nothing but disabling some compiler warnings?!?

**Pull request merged**